src/applications/model/gaming-streaming-server.cc

```diff
@@ -220,36 +218,39 @@ GamingStreamingServer::StartApplication (void)
     }
   m_socket->SetRecvCallback (MakeNullCallback<void, Ptr<Socket> > ());
   m_socket->SetAllowBroadcast (true);
-  for (size_t i = 0; i < m_packetSizeVariables.size (); ++i)
+  for (size_t i = 0; i < m_trafficStreams.size (); ++i)
     {
-      m_sendEvent[i] = Simulator::ScheduleNow (&GamingStreamingServer::Send, this, (int)i);
+      ScheduleNextTx (i);
     }
 }
 
 void
 GamingStreamingServer::StopApplication ()
 {
   NS_LOG_FUNCTION (this);
-  for (EventId sendEvent : m_sendEvent)
+  for (TrafficStream traffic : m_trafficStreams)
```

Comment:
According to the document attached (Email broadcasted by Mattia), `traffic` here should be a copy of the real object in `m_trafficStreams`. Therefore all the subsequent operations are not conducted on the actual object we want to modify
[Text.zip](https://github.com/signetlabdei/ns3-802.11ad/files/4615376/Text.zip)

Reply:
Please check all the for-cycles you wrote using this syntax. It is a nice and modern construct but it's quite dangerous. 
I prefer the use of iterators, for example.